sympy/sets/fancysets.py

```diff
@@ -621,13 +619,14 @@ class ComplexRegion(Set):
     is_ComplexRegion = True
 
     def __new__(cls, sets, polar=False):
-        from sympy import symbols, Dummy
+        from sympy import symbols, Dummy, sympify
 
         x, y, r, theta = symbols('x, y, r, theta', cls=Dummy)
         I = S.ImaginaryUnit
+        polar = sympify(polar)
 
         # Rectangular Form
-        if polar is False:
+        if polar == False:
```

Comment:
For this type of situations, i try to introduce myself to whether any other value apart from `True`, `False` i.e apart from booleans values, how is that going to show behavior.
In this case i expect the user to actually enter only two possible values i.e `True` or `False` and not other value actually. So for instance the user enters `polar=None`, then `if not polar:` is going to `True` and the code block under it will be executed (which we don't want) . Even if the user enters `polar="apple"` then for `if polar:` ( "not is not here") is going to be `True` then the code-block under it will be executed (which again we don't want at all, i guess).

Perhaps i forgot one thing, i.e `raise ValueError` should be introduced for `polar != True or False`, otherwise it will raise `Unbound local error for polar=None`.